src/install/lock.ts

```diff
@@ -288,7 +290,16 @@ export async function extractLockConstraintsAndMap (map: IImportMap, preloadUrls
           continue;
         }
       }
+
+      // Otherwise if the plain specifier is a reference to the base package
+      // containing the import map, we can safely ignore it, as the tracemap
+      // will always resolve such imports correctly at the top-level:
+      const pcfg = await resolver.getPackageConfig(pkgUrl);
```

Comment:
Rather than `pkgUrl` we should try operate on a version of `getPackageBase` scoped to the first package.json boundary only to match Node.js resolution semantics.